mm2src/coins/z_coin.rs

```diff
@@ -1952,6 +2031,83 @@ impl InitWithdrawCoin for ZCoin {
     }
 }
 
+/// Waits until there are enough _unlocked_ Sapling notes to cover `total_required`.
+async fn wait_for_spendable_balance_impl(
+    selfi: ZCoin,
+    total_required: BigDecimal,
+) -> Result<impl Iterator<Item = SpendableNote>, MmError<GenTxError>> {
+    const MAX_RETRIES: usize = 40;
+    const RETRY_DELAY: f64 = 15.0;
```

Comment:
Did you consider passing `wait_until` similar to other functions? we then get this value from swaps by passing it here https://github.com/KomodoPlatform/komodo-defi-framework/blob/a8f606f325676a6cb65399f6bbbe8abaa2e8a2c0/mm2src/coins/z_coin.rs#L1344-L1354 among other places. This is not required in this PR but maybe as a future enhancement if the current approach still leads to failed swaps. Let's add a todo for now.